connector/jobrunner/runner.py

```diff
@@ -98,19 +98,33 @@
 _logger = logging.getLogger(__name__)
 
 
-def _async_http_get(url):
+def _async_http_get(port, db_name, job_uuid):
+    # Method to set failed job (due to timeout, etc) as pending,
+    # to avoid keeping it as enqueued.
+    def set_job_pending():
+        conn = psycopg2.connect(openerp.sql_db.dsn(db_name))
+        conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+        with closing(conn.cursor()) as cr:
+            cr.execute("UPDATE queue_job SET state=%s, "
+                       "date_enqueued=NULL, date_started=NULL, worker_uuid=NULL "
+                       "WHERE uuid=%s", (PENDING, job_uuid))
```

Comment:
Yes, because if the controller changes the state to `started` and we get a timeout or no response, the jobrunner could reset a running job to `enqueued` (not sure that could really happen but anyway that's safer).